pkg/apis/extensions/validation/validation.go

```diff
@@ -593,3 +593,89 @@ func ValidateScale(scale *extensions.Scale) field.ErrorList {
 
 	return allErrs
 }
+
+// ValidatePodSecurityPolicyName can be used to check whether the given
+// pod security policy name is valid.
+// Prefix indicates this name will be used as part of generation, in which case
+// trailing dashes are allowed.
+func ValidatePodSecurityPolicyName(name string, prefix bool) (bool, string) {
+	return apivalidation.NameIsDNSSubdomain(name, prefix)
+}
+
+func ValidatePodSecurityPolicy(psp *extensions.PodSecurityPolicy) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, apivalidation.ValidateObjectMeta(&psp.ObjectMeta, false, ValidatePodSecurityPolicyName, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, ValidatePodSecurityPolicySpec(&psp.Spec, field.NewPath("spec"))...)
+	return allErrs
+}
+
+func ValidatePodSecurityPolicySpec(spec *extensions.PodSecurityPolicySpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// ensure the user strat has a valid type
+	switch spec.RunAsUser.Type {
+	case extensions.RunAsUserStrategyMustRunAs, extensions.RunAsUserStrategyMustRunAsNonRoot, extensions.RunAsUserStrategyRunAsAny, extensions.RunAsUserStrategyMustRunAsRange:
+	//good types
+	default:
+		supported := []string{string(extensions.RunAsUserStrategyMustRunAs),
```

Comment:
Check out how `sets.NewString` is used with `supportedAccessModes` in `pkg/api/validation/validation.go`.  Consider using that style here.  Switch statement not required, and multiple list of strategies not needed.